src/mlpack/core/data/load_csv.hpp

```diff
@@ -263,28 +396,49 @@ class LoadCSV
     inFile.clear();
     inFile.seekg(0, std::ios::beg);
 
-    auto setCharClass = [&](iter_type const &iter)
+    while (std::getline(inFile, line))
     {
-      std::string str(iter.begin(), iter.end());
-      if (str == "\t")
+      // Remove whitespaces from either side
+      trim(line);
+
+      std::stringstream line_stream;
+      std::string token;
+
+      if(line.size() == 0)
       {
-        str.clear();
+        break;
       }
-      boost::trim(str);
 
-      inout(row, col++) = infoSet.template MapString<T>(std::move(str), row);
-    };
+      line_stream.clear();
+      line_stream.str(line);
 
-    while (std::getline(inFile, line))
-    {
-      // Remove whitespace from either side.
-      boost::trim(line);
+      while(line_stream.good())
+      {
+        if(token == "\t")
+        {
+          token.clear();
+        }
 
-      // Parse the numbers from a line (ex: 1,2,3,4); if the parser finds a
-      // number it will execute the setNum function.
-      const bool canParse = qi::parse(line.begin(), line.end(),
-          stringRule[setCharClass] % delimiterRule);
+        std::getline(line_stream, token, delim);
+	// Remove whitespace from either side
+        trim(token);
 
+        if(token[0] == '"' && token[token.size() - 1] != '"')
+        {
+	  std::string tok = token;
+
+	  while(token[token.size() - 1] != '"')
+	  {
+	    tok += delim;
+	    std::getline(line_stream, token, delim);
+	    tok += token;
+          }
+
+	  token = tok;
+        }
+
+	inout(row, col++) = infoSet.template MapString<T>(std::move(token), row);
```

Comment:
Don't we only need to map a token if `ConvertToken()` fails?  But I don't see any call to `ConvertToken()`.  Maybe I misunderstood something?